File: Node.js
class Node {
  constructor(v, n=null) {
    this.val = v;
    this.nxt = n;
  }

  get value() {
    return this.val;
  }

  set value(v) {
    this.val = v;
  }

  get next() {
    return this.nxt;
  }

  set next(n) {
    if (n instanceof Node) {
      this.nxt = n;
    } else {
      throw new Error('Next must be a Node');
    }
  }
}

module.exports = Node;

File: node.rb
class Node
  attr_reader :next, :value

  def initialize(value, n=nil)
    @value = value
    @next = n
  end

  def next=(n)
    if n.instance_of?(Node)
      @next = n
    else
      raise(ArgumentError, "Next must be a Node")
    end
  end

  def value=(val)
    @value = val
  end
end
